### PR TITLE
Added the ability to add a user or group to the dosadmin role during installation.

### PR DESCRIPTION
**Pull request by michaelvidal24**

### CODE REVIEW

File: Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1 [hunk 1]
```diff
@@ -81,6 +89,165 @@ function Add-DatabaseSecurity($userName, $role, $connString)
...
+}
+
+function Get-Role($name, $grain, $securableItem, $authorizationServiceUrl, $accessToken){
+    $url = "$authorizationServiceUrl/roles/$grain/$securableItem/$name"
+    $role = Invoke-Get -url $url -accessToken $accessToken
+    return $role
+}
+
+function Add-Group($authUrl, $name, $source, $accessToken){
+    $url = "$authUrl/groups"
```
Comment (bsmithvt) [inline Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1]: Are JSON KV pairs not comma delimited?

File: Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1 [hunk 2]
```diff
@@ -81,6 +89,165 @@ function Add-DatabaseSecurity($userName, $role, $connString)
...
+    $accountNameParts = $accountName.Split('\')
+    if($accountNameParts.Count -ne 2){
+        Write-Error "Please enter an account in the form DOMAIN\account. Halting installation." -ErrorAction Stop
+    }
+    $samAccountName = $accountNameParts[1]
+    return $samAccountName
+}
+
+function Add-AccountToDosAdminRole($accountName, $domain, $authorizationServiceURL, $accessToken, $connString){
+    $samAccountName = Get-SamAccountFromAccountName -accountName $accountName
```
Comment (bsmithvt) [inline Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1]: Should the variable passed to the `$authorizationServiceUrl` parameter be `$authorizationServiceURL` and not `$authorizationServiceUrl`?

File: Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1 [hunk 3]
```diff
@@ -325,6 +494,35 @@ Invoke-Sql $authorizationDbConnStr "SELECT TOP 1 ClientId FROM Clients" | Out-Nu
 Write-Success "Identity DB Connection string: $authorizationDbConnStr verified"
 Write-Host ""
 
+$userEnteredDomain = Read-Host "Press Enter to accept the default domain '$($currentUserDomain)' that the user/group who will administrate dos is a member or, or enter a new domain"
```
Comment (bsmithvt) [inline Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1]: I think the error message above has a slight typo.

File: Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1 [hunk 4]
```diff
@@ -325,6 +494,35 @@ Invoke-Sql $authorizationDbConnStr "SELECT TOP 1 ClientId FROM Clients" | Out-Nu
...
+$userEnteredDomain = Read-Host "Press Enter to accept the default domain '$($currentUserDomain)' that the user/group who will administrate dos is a member or, or enter a new domain" 
+Write-Host ""
+
+if(![string]::IsNullOrEmpty($userEnteredDomain)){
+    $currentUserDomain = $userEnteredDomain
+}
+
+if([string]::IsNullOrEmpty($adminAccount)){
+    $userEnteredAdminAccount = Read-Host "Please enter the user/group account for dos administration in the format [DOMAIN\user]"
+}else{
```
Comment (bsmithvt) [inline Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1]: I think the error message has a slight typo.

File: Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1 (hunk 1 above)
Comment (michaelvidal24) [reply Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1]: We are creating a hashtable here in PowerShell, then in the Invoke-Post method, we convert it to Json.

File: Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1 (hunk 2 above)
Comment (michaelvidal24) [reply Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1]: Yeah - I'll fix that up.

**Pull request merged**